Code/src/main/java/com/html5parser/algorithms/InsertCharacter.java

```diff
@@ -60,6 +60,9 @@ else if (adjustedInsertionLocation.getParent().getLastChild() != null
 			Node location = adjustedInsertionLocation.getParent()
 					.getLastChild();
 			location.setNodeValue((location.getNodeValue()).concat(data));
+			if (location.getNodeValue().contains("&amp;")) {
```

Comment:
This does not fix the real bug. Is just a patch. Actually the error is in GenericRCDATAElementParsing algorithm